server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationStatsIT.java

```diff
@@ -22,33 +22,43 @@
 import static java.util.Arrays.asList;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
 
 @OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
 public class SegmentReplicationStatsIT extends SegmentReplicationBaseIT {
 
-    public void testSegmentReplicationStatsResponse() throws Exception {
-        final String primaryNode = internalCluster().startNode();
-        createIndex(INDEX_NAME);
-        ensureYellowAndNoInitializingShards(INDEX_NAME);
-        final String replicaNode = internalCluster().startNode();
-        ensureGreen(INDEX_NAME);
-
-        // index 10 docs
-        for (int i = 0; i < 10; i++) {
-            client().prepareIndex(INDEX_NAME).setId(Integer.toString(i)).setSource("field", "value" + i).execute().actionGet();
+    public void testSegmentReplicationStatsResponse() {
```

Comment:
@Rishikesh1159 : I fixed this flaky test in https://github.com/opensearch-project/OpenSearch/pull/6370. Can you please rebase your changes against `main`?

Reply:
Sure let me rebase

Reply:
FYI, I see one run where this is still flaky https://github.com/opensearch-project/OpenSearch/pull/6366#issuecomment-1437664743. 
Previously, I root caused the problem happening due to race condition and fixed by asserting on replication state to be done which fixed it from 20% failing -> 0%. The assertBusy block by default waits for `10s` not sure if segment replication needs more time to complete. May be we can increase the assertBusy timeout to 60secs if this test is still flaky.

Reply:
Yes testSegmentReplicationStatsResponse() was flaky previously because of stage not reaching done within certain time. But now I have changed the test completely. As we are only testing API response and we are concerned only about API response but not about segment replication stage. So I changed the test and we no more assert on segrep stage which was causing flakiness.

Reply:
With changes in testSegmentReplicationStatsResponse() we only assert if we get a successful or failure response from API call. We are not concerned about if segment replication event has finished processing or not, as urpose of API is only to give response irrespective of state of segrep event.

Reply:
I think there is still value in asserting the replication state as it can uncover bugs where replication never completes for various reasons and it is also part of API response. I will leave it for you to decide.

Reply:
Sure, I can add asserts on replication state. It doesn't break anything but as you said there is a chance it can uncover some bugs. I have added that in latest commit. Thanks @dreamer-89